### ISSUE TITLE
False-positive RedundantCondition: count on non-empty-array

### ISSUE THREAD

**Issue by sserbin** [vimeo/psalm#1140]
```php
<?php
  
function foo(string $s) : void {
    $segments = explode('.', $s);
    if (count($segments) === 1) {
    	// do something, early return
    }
}

foo("a.b.c");
```
https://getpsalm.org/r/b345dc049e

Expected: no errors
Actual: `ERROR: RedundantCondition - 5:9 - Found a redundant condition when evaluating $segments and trying to reconcile type 'non-empty-array<int, string>' to non-empty-countable`

**Comment by muglug** [comment]
Sorry - I accidentally, while working on a new feature, committed to the master branch. I've reverted the two commits.

**Comment by muglug** [comment]
The feature has since been released in master, btw
